cosmos/lib/cosmos/interfaces/stream_interface.rb

```diff
@@ -55,13 +55,19 @@ def disconnect
     end
 
     def read_interface
+      timeout = false
       begin
         data = @stream.read
       rescue Timeout::Error
         Logger.instance.error "#{@name}: Timeout waiting for data to be read"
+        timeout = true
         data = nil
       end
-      return nil if data.nil? or data.length <= 0
+      if data.nil? or data.length <= 0
+        Logger.instance.info "#{@name}: #{@stream.class} read returned nil" if data.nil? and not timeout
+        Logger.instance.info "#{@name}: #{@stream.class} read returned 0 bytes (stream closed)" if not data.nil? and data.length <= 0
```

Comment:
So you get this message and then the one above which says Interface read_interface requested disconnect. I think that makes sense.